client.go

```diff
@@ -21,74 +21,168 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"strings"
 
 	"github.com/mendersoftware/log"
 )
 
 var (
-	errorLoadingClientCertificate = errors.New("Failed to load certificate and key")
-	errorNoServerCertificateFound = errors.New("No server certificate is provided," +
-		" use -trusted-certs with a proper certificate.")
-	errorAddingServerCertificateToPool = errors.New("Adding server certificate " +
-		"to trusted pool failed.")
+	errorLoadingClientCertificate      = errors.New("Failed to load certificate and key")
+	errorNoServerCertificateFound      = errors.New("No server certificate is provided, use -trusted-certs with a proper certificate.")
+	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
-//TODO: this will be hardcoded for now but should be configurable in future
 const (
-	defaultCertFile         = "/data/certfile.crt"
-	defaultCertKey          = "/data/certkey.key"
-	defaultServerCert       = "/data/server.crt"
-	minimumImageSize  int64 = 4096
+	minimumImageSize int64 = 4096 //kB
 )
 
 type RequestProcessingFunc func(response *http.Response) (interface{}, error)
 
 type Updater interface {
-	GetScheduledUpdate(RequestProcessingFunc, string) (interface{}, error)
+	GetScheduledUpdate(RequestProcessingFunc, string, string) (interface{}, error)
 	FetchUpdate(string) (io.ReadCloser, int64, error)
 }
 
 // Client represents the http(s) client used for network communication.
 //
-type client struct {
-	authCredsType
+type httpClient struct {
 	HTTPClient   *http.Client
 	minImageSize int64
 }
 
+type httpsClient struct {
+	httpClient
+	httpsClientAuthCreds
+}
+
 // Client initialization
 
-func NewClient(args authCmdLineArgsType) *client {
-	var httpsClient client
-	httpsClient.minImageSize = minimumImageSize
-	args.setDefaultKeysAndCerts(defaultCertFile, defaultCertKey, defaultServerCert)
+func NewUpdater(conf httpsClientConfig) (Updater, error) {
 
-	if err := httpsClient.initServerTrust(args); err != nil {
-		return nil
+	if conf == (httpsClientConfig{}) {
```

Comment:
this compare is a bit risky, if it has any references then it will compare addresses (not values), even if it does not have it now, if added later it could potentially start misbehaving 

alternatife is reflext.DeepEqual but this is known to be expensive call so have to be used wisely